overlord/snapstate/handlers.go

```diff
@@ -1432,6 +1432,10 @@ func (m *SnapManager) doUnlinkCurrentSnap(t *state.Task, _ *tomb.Tomb) (err erro
 		return err
 	}
 
+	if err := saveCurrentKernelModuleComponents(t, snapsup, snapst); err != nil {
```

Comment:
Aren't the unlink-current-component tasks running before unlink-current-snap and therefore the state has mutated already?

Reply:
how do we test for this in general, do we need variant of many install/refresh tests that involve specifically kernel components and not just components in general?

Reply:
Many of the `testInstallComponentsRunThrough` and `testUpdateWithComponentsRunThrough` variants include kernel-modules components, and test that we end up setting up/undoing set up of the correct kernel modules. I also added to the specific task handler tests cases for testing that we set the correct thing on the SnapSetup.

@alfonsosanchezbeato `unlink-current-component` will only run if we are replacing existing components, if we aren't doing that then we need to capture the list when we're unlinking the snap.

Reply:
ok, I see